examples/private-cluster/main.tf

```
@@ -357,3 +369,70 @@ module "s3_bucket" {
 
   tags = local.tags
 }
+
+resource "aws_iam_role" "custom_instance_profile" {
+  name               = "custom-instance-profile"
+  assume_role_policy = data.aws_iam_policy_document.assume.json
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      identifiers = ["ec2.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "emr_for_ec2" {
+  role       = aws_iam_role.custom_instance_profile.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonElasticMapReduceforEC2Role"
+}
+
+resource "aws_iam_instance_profile" "custom_instance_profile" {
+  role = aws_iam_role.custom_instance_profile.name
+
+  name = "custom-instance-profile"
+
+  depends_on = [
+    aws_iam_role_policy_attachment.emr_for_ec2,
+  ]
+}
+
+resource "aws_iam_role" "autoscaling" {
+  name               = "custom-autoscaling-role"
```

```suggestion
  name_prefix        = "custom-autoscaling-role"
```
Just to avoid conflicts if such a name already exists in the AWS account.